lib/src/query/property.dart

```diff
@@ -0,0 +1,227 @@
+part of query;
+
+class PropertyQuery {
+  Pointer<Void> _cProp;
+  int _type;
+  bool _distinct;
+
+  PropertyQuery(Pointer<Void> cQuery, int propertyId, int obxType) {
+    this._type = obxType;
+    _cProp = checkObxPtr(bindings.obx_query_prop(cQuery, propertyId), "property query");
+  }
+
+  void close() {
+    checkObx(bindings.obx_query_prop_close(_cProp));
+  }
+
+  get distinct => _distinct;
+
+  set distinct (bool d) {
+    _distinct = d;
+    checkObx(bindings.obx_query_prop_distinct(_cProp, d ? 1 : 0));
+  }
+
+  int count() {
+    final ptr = allocate<Uint64>(count: 1);
+    try {
+      checkObx(bindings.obx_query_prop_count(_cProp, ptr));
+      return ptr.value;
+    }finally {
+      free(ptr);
+    }
+  }
+
+  Pointer<TStruct> _curryWithDefault<TStruct extends Struct, N extends NativeType>
+      (obx_query_prop_find_t<TStruct, N> findFn, Pointer<N> cDefault, String errorMessage) {
+    try {
+      return checkObxPtr(findFn(_cProp, cDefault), errorMessage);
+    }finally {
+      if (cDefault.address != 0) {
+        free(cDefault);
+      }
+    }
+  }
+}
+
+/// shared implementation, hence mixin
+mixin _CommonNumeric on PropertyQuery {
+
+  double average() {
+    final ptr = allocate<Double>();
+    checkObx(bindings.obx_query_prop_avg(_cProp, ptr));
+    try {
+      return ptr.value;
+    }finally {
+      free(ptr);
+    }
+  }
+
+}
+
+class IntegerPropertyQuery extends PropertyQuery with _CommonNumeric {
+
+  IntegerPropertyQuery (Pointer<Void> query, int propertyId, int obxType): super(query, propertyId, obxType);
+
+  int _op(obx_query_prop_op_t<int, Int64> fn) {
+    final ptr = allocate<Int64>();
+    checkObx(fn(_cProp, ptr));
+    try {
+      return ptr.value;
+    }finally {
+      free(ptr);
+    }
+  }
+
+  int min() {
+    return _op(bindings.obx_query_prop_min_int);
+  }
+  int max() {
+    return _op(bindings.obx_query_prop_max_int);
+  }
+  int sum() {
+    return _op(bindings.obx_query_prop_sum_int);
+  }
+
+  List<int> _unpack8(Pointer<OBX_int8_array> ptr) {
+    try {
+      return ptr.ref.items();
+    } finally {
+      bindings.obx_int8_array_free(ptr);
+    }
+  }
+
+  List<int> _unpack16(Pointer<OBX_int16_array> ptr) {
+    try {
+      return ptr.ref.items();
+    } finally {
+      bindings.obx_int16_array_free(ptr);
+    }
+  }
+
+  List<int> _unpack32(Pointer<OBX_int32_array> ptr) {
+    try {
+      return ptr.ref.items();
+    } finally {
+      bindings.obx_int32_array_free(ptr);
+    }
+  }
+
+  List<int> _unpack64(Pointer<OBX_int64_array> ptr) {
+    try {
+      return ptr.ref.items();
+    }finally {
+      bindings.obx_int64_array_free(ptr);
+    }
+  }
+
+  List<int> find({int replaceNullWith}) {
+    final ptr = replaceNullWith != null ? (allocate<Int64>()..value = replaceNullWith) : Pointer<Int64>.fromAddress(0);
```

Comment:
looks like another memory leak

Reply:
The pointer is freed within `_curryWithDefault` protected against `checkObxPtr` throwing, so fine?